app/models/course.rb

```diff
@@ -27,17 +27,6 @@ class Course < ApplicationRecord
   audited
 
   attribute :subordinate_subject_id, :integer
-  after_initialize do
```

Comment:
Everytime a Course was instantiate we were doing more queries to the DB so this was necessary to be moved. I moved the methods more below and didn't a way that all the specs passed.